src/Codeception/Lib/Driver/Db.php

```diff
@@ -86,19 +113,37 @@ public static function getProvider($dsn)
      * @param $user
      * @param $password
      * @param [optional] $options
+     * @param [optional] $pdo_class
      *
      * @see http://php.net/manual/en/pdo.construct.php
      * @see http://php.net/manual/de/ref.pdo-mysql.php#pdo-mysql.constants
      */
-    public function __construct($dsn, $user, $password, $options = null)
+    public function __construct($dsn, $user, $password, $options = null, $pdo_class = null)
     {
-        $this->dbh = new \PDO($dsn, $user, $password, $options);
+        $class_name = self::pdoClass($pdo_class);
+        $this->dbh = new $class_name($dsn, $user, $password, $options);
+        self::assertIsPdo($this->dbh, $pdo_class);
         $this->dbh->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
 
         $this->dsn = $dsn;
         $this->user = $user;
         $this->password = $password;
         $this->options = $options;
+        $this->pdo_class = $pdo_class;
```

Comment:
Storing this the same as the rest so it can be passed back in if necessary (e.g. sqllite reconnection)